packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/container.tsx

```diff
@@ -35,16 +40,83 @@ const SignupConfirmCodeContainer = ({
     keyFetchToken,
     unwrapBKey,
   } = location.state || {};
+  const navigate = useNavigate();
+
+  const storedLocalAccount: StoredAccountData | undefined = currentAccount();
+  const { email, sessionToken } = storedLocalAccount || {};
 
   const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
     authClient,
     integration
   );
 
-  if (!integration) {
-    return (
-      <LoadingSpinner className="bg-grey-20 flex items-center flex-col justify-center h-screen select-none" />
-    );
+  const navigateToContentServer = useCallback(
+    (path: string, hasBounced?: boolean) => {
+      const params = new URLSearchParams(location.search);
+
+      // Passing back the 'email' param causes various behaviors in
+      // content-server since it marks the email as "coming from a RP".
+      // Also remove `emailFromContent` since we pass that when coming
+      // from content-server to Backbone, see Signup container component
+      // for more info.
+      params.delete('emailFromContent');
+      params.delete('email');
+      // passing the 'bouncedEmail' param will display an error tooltip
+      // on the email-first signin/signup page and allow to check
+      // if the entered email matches the bounced email
+      hasBounced && params.set('bouncedEmail', email);
+      if (Array.from(params).length > 0) {
+        path += `?${params.toString()}`;
+      }
+      hardNavigateToContentServer(path);
+    },
+    [email, location.search]
+  );
+
+  // Poll for hard bounces registered in database for the entered email.
+  // Previously, we checked if the account was deleted, and assumed
+  // that implied the email bounced/was invalid.
+  const { data } = useQuery<GetEmailBounceStatusResponse>(
+    EMAIL_BOUNCE_STATUS_QUERY,
+    {
+      variables: { input: email },
+      pollInterval: POLL_INTERVAL,
+    }
+  );
+
+  // Handle email bounces
+  useEffect(() => {
+    if (data?.emailBounceStatus.hasHardBounce) {
+      const hasBounced = true;
+      // if arriving from signup, return to '/' and allow user to signup with another email
+      if (
+        location.state &&
+        'origin' in location.state &&
+        location.state.origin === 'signup'
+      ) {
+        navigateToContentServer('/', hasBounced);
+      } else {
+        // if not arriving from signup, redirect to signin_bounced for support info
+        navigateToContentServer('/signin_bounced', hasBounced);
+      }
+    }
+  }, [data, location, navigate, navigateToContentServer]);
+
+  if (!integration || !sessionToken) {
```

Comment:
Same question as this: https://github.com/mozilla/fxa/pull/16175#discussion_r1430793333

I think we want a loading spinner for the `(!integration)` check and not navigate users away.